lib/index.js

```diff
@@ -1,11 +1,19 @@
 'use strict';
 
+const RateLimiter = require('./rate-limiter');
+
 exports.register = (server, options, next) => {
 
   server.ext('onPostAuth', (request, reply) => {
-    request.payload.loaded = true;
+    return RateLimiter.limit(request, options.redisClient, options.defaultRate)
+    .then((rate) => {
+      if (rate && rate.current > rate.limit) {
+        return reply(options.overLimitHandler());
```

Comment:
I think we should start with the simpler case and just throw the error since that's all we need it to do. If we find that we actually do need the flexibility, we can always change it to take in a handler function.